### PR TITLE
13021 - Changes for Special Resolution Draft

### PR DESCRIPTION
**Pull request by seeker25**
*Issue #:* /bcgov/entity#13021

*Description of changes:*
Changes for Special Resolution Draft

Waiting on:

https://github.com/bcgov/bcrs-shared-components/pull/146

Also need to do a package-lock update

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/TodoList.vue [hunk 1]
```diff
@@ -1370,6 +1415,13 @@ export default class TodoList extends Mixins(
         navigate(conversionUrl)
         break
 
+      case FilingTypes.SPECIAL_RESOLUTION:
+        // navigate to Edit UI to resume this special resolution
+        // const specialResolutionUrl = `${this.editUrl}${this.getIdentifier}/special-resolution/?special-resolution-id=${item.filingId}`
+        const specialResolutionUrl = `http://localhost:8081/basePath/${this.getIdentifier}/special-resolution/?special-resolution-id=${item.filingId}`
```
Comment (seeker25) [inline src/components/Dashboard/TodoList.vue:1421]: This need to be moved back before merging

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-filings-ui",
-  "version": "5.3.5",
+  "version": "5.3.6",
```
Comment (seeker25) [inline package.json:3]: Package lock needs to be updated, after the packages are created for `bcrs-shared-components`

File: package.json [hunk 3]
```diff
@@ -16,14 +16,14 @@
...
     "@bcrs-shared-components/corp-type-module": "1.0.6",
     "@bcrs-shared-components/court-order-poa": "1.0.14",
-    "@bcrs-shared-components/enums": "1.0.14",
-    "@bcrs-shared-components/interfaces": "1.0.37",
-    "@bcrs-shared-components/staff-comments": "1.2.0",
-    "@bcrs-shared-components/staff-payment": "1.1.0",
+    "@bcrs-shared-components/enums": "1.0.20",
+    "@bcrs-shared-components/interfaces": "1.0.40",
+    "@bcrs-shared-components/staff-comments": "1.2.6",
+    "@bcrs-shared-components/staff-payment": "2.0.10",
```
Comment (seeker25) [inline package.json:26]: Major change, testing this now.

**Review by cameron-freshworks — APPROVED** [review]
Code looks good to me!

File: package.json (hunk 3 above)
Comment (seeker25) [reply package.json:26]: I did a Director change, paid with a routing slip.. seemed to work.

Comment (severinbeauvais) [reply package.json:26]: Thanks for checking! You could also do a diff (via GH history) of v1 versus v2 and target what you're testing.

Usually a major version change means something will be incompatible.

File: package.json (hunk 2 above)
Comment (severinbeauvais) [inline package.json:3]: Given all the import changes, I suggest you change this to 5.4.0 :)

File: src/components/Dashboard/TodoList.vue [hunk 4]
```diff
@@ -1272,6 +1278,45 @@ export default class TodoList extends Mixins(
...
+        paymentMethod: header.paymentMethod || null,
+        paymentToken: header.paymentToken || null,
+        payErrorObj,
+        comments: this.flattenAndSortComments(header.comments),
+        commentsLink: `businesses/${this.getIdentifier}/filings/${header.filingId}/comments`
+      }
+      this.todoItems.push(item)
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('ERROR - invalid header or specialResolution or business in task =', task)
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:1316]: You're not checking if business exists.

File: src/components/Dashboard/TodoList.vue [hunk 5]
```diff
@@ -1272,6 +1278,45 @@ export default class TodoList extends Mixins(
...
 
+  private async loadSpecialResolution (task: ApiTaskIF): Promise<void> {
+    const filing = task.task.filing
+    const header = filing.header
+    const business = filing.business
+    const specialResolution = filing.specialResolution
+
+    if (header && specialResolution) {
+      if (!this.isGoodStanding && this.isStatusDraft(header)) {
+        task.enabled = false
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:1289]: Well, you _shouldn't_ change the original data, but I know it's not used for anything else so not a big deal.

Comment (seeker25) [reply src/components/Dashboard/TodoList.vue:1289]: I borrowed this from Alteration

File: src/components/Dashboard/TodoList.vue (hunk 4 above)
Comment (seeker25) [reply src/components/Dashboard/TodoList.vue]: fixed

File: src/interfaces/index.ts [hunk 6]
```diff
@@ -2,16 +2,16 @@
...
 // https://devblogs.microsoft.com/typescript/announcing-typescript-3-8-beta/
 import type { CorpInfoIF } from '@bcrs-shared-components/corp-type-module'
-import type { AlterationIF, ContactPointIF, NameRequestIF, NameTranslationIF, NameRequestDetailsIF,
-  NameRequestApplicantIF, ShareStructureIF, CourtOrderIF, IncorporationApplicationIF,
-  IncorporationAddressIf, ShareClassIF, CommentIF, ConfirmDialogType, StaffPaymentIF, ApiDateTimeUtc,
-  FormattedDateTimeGmt, IsoDatePacific } from '@bcrs-shared-components/interfaces'
+import type { AlterationIF, ChangeOfNameIF, ContactPointIF, NameRequestIF,
+  NameTranslationIF, NameRequestDetailsIF, NameRequestApplicantIF, ShareStructureIF, CourtOrderIF,
+  IncorporationApplicationIF, IncorporationAddressIf, ShareClassIF, CommentIF, ConfirmDialogType, StaffPaymentIF,
+  ApiDateTimeUtc, FormattedDateTimeGmt, IsoDatePacific, SpecialResolutionIF } from '@bcrs-shared-components/interfaces'
```
Comment (severinbeauvais) [inline src/interfaces/index.ts:8]: Did you move any of these interfaces from this project to bcrs-shared-components? (And if so, can they be deleted from this project now?)

File: src/components/Dashboard/TodoList.vue (hunk 5 above)
Comment (severinbeauvais) [reply src/components/Dashboard/TodoList.vue:1289]: Doh. Just leave it then 😋

File: src/interfaces/index.ts (hunk 6 above)
Comment (seeker25) [reply src/interfaces/index.ts:8]: I don't think so, mostly from `business-edit-ui`

Comment (seeker25) [reply src/interfaces/index.ts:8]: I honestly started refactoring as much as I could.. but realized I didn't want to go too crazy on the first time working on these repos. 

Also you probably would have told me to create a separate PR for that work.

Comment (severinbeauvais) [reply src/interfaces/index.ts:8]: Sounds good. Thanks for improving this project!

File: package.json (hunk 2 above)
Comment (severinbeauvais) [reply package.json]: Outdated conversation. Resolving it.

**Review by severinbeauvais — APPROVED** [review]
The code changes look good to me, too.

**Pull request merged**